shell/platform/common/cpp/text_input_model.cc

```diff
@@ -113,6 +119,42 @@ bool TextInputModel::Delete() {
   return false;
 }
 
+bool TextInputModel::DeleteSurrounding(int offset, int n_chars) {
+  auto start = selection_base_;
```

Comment:
The cursor is selection_offset_, not selection_base_, unless I'm misremembering.